File: frontend/src/api/baidu.ts
import { invoke } from '@tauri-apps/api/tauri'

export const baiduApi = {
  // 获取访问令牌
  async getToken(code: string, clientId: string, clientSecret: string, redirectUri: string) {
    return invoke('baidu_get_token', { code, clientId, clientSecret, redirectUri })
  },
  
  // 刷新令牌
  async refreshToken(refreshToken: string, clientId: string, clientSecret: string) {
    return invoke('baidu_refresh_token', { refreshToken, clientId, clientSecret })
  },
  
  // 验证令牌
  async verifyToken(accessToken: string) {
    return invoke('baidu_verify_token', { accessToken })
  },
  
  // 获取文件列表
  async getFileList(
    accessToken: string,
    dir: string,
    pageNum: number = 1,
    pageSize: number = 100,
    order: string = 'name',
    method: string = 'list',
    recursion: number = 0
  ) {
    return invoke('baidu_list_files', {
      accessToken,
      dir,
      pageNum,
      pageSize,
      order,
      method,
      recursion
    })
  },
  
  // 搜索文件
  async searchFiles(
    accessToken: string,
    key: string,
    dir: string,
    method: string = 'search',
    recursion: number = 1
  ) {
    return invoke('baidu_search_files', {
      accessToken,
      key,
      dir,
      method,
      recursion
    })
  },
  
  // 获取文件信息
  async getFileInfo(accessToken: string, fsids: string) {
    return invoke('baidu_get_fileinfo', { accessToken, fsids })
  },
  
  // 下载文件
  async downloadFile(dlink: string, accessToken: string) {
    return invoke('baidu_download_file', { dlink, accessToken })
  },
  
  // 上传文件
  async uploadFile(fileName: string, fileData: number[], accessToken: string) {
    return invoke('baidu_upload_file', { fileName, fileData, accessToken })
  }
}

File: src-tauri/src/commands/baidu.rs
use crate::api::baidu::BaiduClient;
use crate::api::types::BaiduTokenResponse;
use serde_json::Value;

/// Get Baidu access token
#[tauri::command]
pub async fn baidu_get_token(
    code: String,
    client_id: String,
    client_secret: String,
    redirect_uri: String,
) -> Result<BaiduTokenResponse, String> {
    let client = BaiduClient::new();
    client
        .get_token(&code, &client_id, &client_secret, &redirect_uri)
        .await
        .map_err(|e| e.to_string())
}

/// Refresh Baidu access token
#[tauri::command]
pub async fn baidu_refresh_token(
    refresh_token: String,
    client_id: String,
    client_secret: String,
) -> Result<BaiduTokenResponse, String> {
    let client = BaiduClient::new();
    client
        .refresh_token(&refresh_token, &client_id, &client_secret)
        .await
        .map_err(|e| e.to_string())
}

/// Verify Baidu access token
#[tauri::command]
pub async fn baidu_verify_token(access_token: String) -> Result<Value, String> {
    let client = BaiduClient::new();
    client
        .verify_token(&access_token)
        .await
        .map_err(|e| e.to_string())
}

/// List files in Baidu Netdisk
#[tauri::command]
pub async fn baidu_list_files(
    access_token: String,
    dir: String,
    page_num: Option<u32>,
    page_size: Option<u32>,
    order: Option<String>,
    recursion: Option<u8>,
) -> Result<Value, String> {
    let client = BaiduClient::new();
    client
        .list_files(
            &access_token,
            &dir,
            page_num,
            page_size,
            order.as_deref(),
            recursion,
        )
        .await
        .map_err(|e| e.to_string())
}

/// Search files in Baidu Netdisk
#[tauri::command]
pub async fn baidu_search_files(
    access_token: String,
    key: String,
    dir: String,
    recursion: Option<u8>,
) -> Result<Value, String> {
    let client = BaiduClient::new();
    client
        .search_files(&access_token, &key, &dir, recursion)
        .await
        .map_err(|e| e.to_string())
}

/// Upload file to Baidu Netdisk
#[tauri::command]
pub async fn baidu_upload_file(
    access_token: String,
    file_name: String,
    file_data: Vec<u8>,
) -> Result<Value, String> {
    let client = BaiduClient::new();
    client
        .upload_file(&file_name, file_data, &access_token)
        .await
        .map_err(|e| e.to_string())
}

/// Download file from Baidu Netdisk
#[tauri::command]
pub async fn baidu_download_file(
    dlink: String,
    access_token: String,
) -> Result<Vec<u8>, String> {
    let client = BaiduClient::new();
    client
        .download_file(&dlink, &access_token)
        .await
        .map_err(|e| e.to_string())
}

/// Get file info from Baidu Netdisk
#[tauri::command]
pub async fn baidu_get_fileinfo(
    access_token: String,
    fsids: String,
) -> Result<Value, String> {
    let client = BaiduClient::new();
    client
        .get_fileinfo(&access_token, &fsids)
        .await
        .map_err(|e| e.to_string())
}
